File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        int len = 0;
        queue<TreeNode*> queue;
        TreeNode* current;
        vector<vector<int>> res;
        vector<int> sub;
        if (root)
        {
            queue.push(root);
        }
        while (queue.size() > 0)
        {
            len = queue.size();
            for (int i = 0; i < len; i++)
            {
                current = queue.front();
                queue.pop();
                sub.push_back(current->val);
                if (current->left)
                    queue.push(current->left);
                if (current->right)
                    queue.push(current->right);
            }
            res.push_back(sub);
            sub.clear();
        }
        return (res);
    }
};

File: Data Structures & Algorithms/invert-a-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        queue = deque()
        if root:
            queue.append(root)
        while len(queue) > 0:
            size = len(queue)
            for i in range(size):
                current = queue[0]
                queue.popleft()
                if current.left:
                    queue.append(current.left)
                if current.right:
                    queue.append(current.right)
                swap = current.left
                current.left = current.right
                current.right = swap
        return root